---
title: Lint Rule js/noDebugger
layout: layouts/page.njk
description: MISSING DOCUMENTATION
eleventyNavigation: {
	key: lint-rules/js/noDebugger,
	parent: lint-rules,
	title: js/noDebugger
}
---

# js/noDebugger

MISSING DOCUMENTATION

<!-- EVERYTHING BELOW IS AUTOGENERATED. SEE SCRIPTS FOLDER FOR UPDATE SCRIPTS -->


## Examples
## Invalid
```typescript
debugger;
```
## Valid
```typescript
const test = { debugger: 1 };
test.debugger;
```
