---
title: Lint Rule js/noDuplicateCase
layout: layouts/page.njk
description: MISSING DOCUMENTATION
eleventyNavigation: {
	key: lint-rules/js/noDuplicateCase,
	parent: lint-rules,
	title: js/noDuplicateCase
}
---

# js/noDuplicateCase

MISSING DOCUMENTATION

<!-- EVERYTHING BELOW IS AUTOGENERATED. SEE SCRIPTS FOLDER FOR UPDATE SCRIPTS -->


## Examples
## Invalid
```typescript
const expr = 'a';
switch (expr) {
	case 'a':
		break;
	case 'b':
		break;
	case 'c':
		break;
	case 'd':
		break;
	case 'c':
		break;
	default:
		break;
}
```
```typescript
const expr = 3;
switch (expr) {
	case 1:
		break;
	case 2:
		break;
	case 3:
		break;
	case 2:
		break;
	default:
		break;
}
```
```typescript
const expr = 3;
switch (expr) {
	case 1:
		break;
	case 2n:
		break;
	case 3:
		break;
	case 2n:
		break;
	default:
		break;
}
```
```typescript
const foo = 'a';
switch ('a') {
	case foo:
		break;
	case foo:
		break;
	default:
		break;
}
```
```typescript
const foo = 'a';
switch ('a') {
	case 'foo':
		break;
	case 'foo':
		break;
	default:
		break;
}
```
```typescript
const foo = 'a';
switch ('a') {
	case null:
		break;
	case null:
		break;
	default:
		break;
}
```
## Valid
```typescript
const expr = 'a';
switch (expr) {
	case 'a':
		break;
	case 'b':
		break;
	case 'c':
		break;
	case 'd':
		break;
	default:
		break;
}
```
```typescript
const expr = 3;
switch (expr) {
	case 1:
		break;
	case 2:
		break;
	case 3:
		break;
	case 2n:
		break;
	default:
		break;
}
```
```typescript
const expr = 3;
switch (expr) {
	case 1:
		break;
	case '\n1':
		break;
	case 1n:
		break;
	case 'null':
		break;
	case null:
		break;
	case foo:
		break;
	case 'foo':
		break;
	default:
		break;
}
```
