---
title: Lint Rule js/noLabelVar
layout: layouts/page.njk
description: MISSING DOCUMENTATION
eleventyNavigation: {
	key: lint-rules/js/noLabelVar,
	parent: lint-rules,
	title: js/noLabelVar
}
---

# js/noLabelVar

MISSING DOCUMENTATION

<!-- EVERYTHING BELOW IS AUTOGENERATED. SEE SCRIPTS FOLDER FOR UPDATE SCRIPTS -->


## Examples
## Invalid
```typescript
const x = 'test';
x: expr;
```
## Valid
```typescript
const x = 'test';
z: expr;
```
