---
title: Lint Rule js/noReferenceToNonExistingGroup
layout: layouts/page.njk
description: MISSING DOCUMENTATION
eleventyNavigation:
	key: lint-rules/js/noReferenceToNonExistingGroup
	parent: lint-rules
	title: js/noReferenceToNonExistingGroup
---

# js/noReferenceToNonExistingGroup

MISSING DOCUMENTATION

<!-- EVERYTHING BELOW IS AUTOGENERATED. SEE SCRIPTS FOLDER FOR UPDATE SCRIPTS -->


## Examples
## Invalid
```typescript
String.raw`let foo = /([abc]+)=18/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=49/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=78/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=99/;foo;`
```
```typescript
String.raw`let foo = /(([abc])19)+=28/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=199/;foo;`
```
## Valid
```typescript
String.raw`let foo = /([abc]+)=1/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=2/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=8/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=9/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=119/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=338/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=377/;foo;`
```
```typescript
String.raw`let foo = /([abc]+)=777/;foo;`
```
