---
title: Lint Rule js/preferTemplate
layout: layouts/page.njk
description: MISSING DOCUMENTATION
eleventyNavigation: {
	key: lint-rules/js/preferTemplate,
	parent: lint-rules,
	title: js/preferTemplate
}
---

# js/preferTemplate

MISSING DOCUMENTATION

<!-- EVERYTHING BELOW IS AUTOGENERATED. SEE SCRIPTS FOLDER FOR UPDATE SCRIPTS -->


## Examples
## Invalid
```typescript
const foo = 'bar'; console.log(foo + 'baz')
```
```typescript
console.log((1 * 2) + 'foo')
```
```typescript
console.log(1 + 'foo' + 2 + 'bar' + 'baz' + 3)
```
```typescript
console.log((1 + 'foo') * 2)
```
```typescript
console.log((1 * (2 + 'foo')) + 'bar')
```
```typescript
console.log('foo' + 1)
```
```typescript
console.log('foo' + `bar${`baz${'bat' + 'bam'}`}` + 'boo')
```
```typescript
console.log('foo' + 1 + 2)
```
## Valid
```typescript
console.log('foo' + 'bar')
```
```typescript
console.log(foo() + '\n')
```
